### PR TITLE
Make SmartLedsWrite more generic

### LINKED ISSUE #3: Support 16bit color per channel
Some LED strips (like APA102 with global brightness bits) can achieve greater than 8-bit per color channel resolution under certain conditions. Similarly LEDs driven using 12-bit or 16-bit pwm (either via peripherals or an external PWM chip) would also be nice to support.

For context I'm working on a crate that implements the interpolation, gamma correction and temporal dithering techniques from fadecandy. It supports writing into the internal state using the `SmartLedsWrite` trait and itself writes to the underlying LEDs via the same. I have a few custom LED spotlights that use 16-bit pwm and would like to re-use the interpolation and gamma correction to take advantage of the 16-bit pwm during fades while still only having to provide 8-bit per channel input.

It seems like either a new 16-bit specific trait could be defined or now that this is using the rgb crate  a generic trait over `RGB` could be defined. e.g.

```
pub use rgb::RGB;

pub type Color = rgb::RGB8;
pub type Color16 = rgb::RGB16;

pub trait SmartLedsWrite<RGB> {
    type Error;
    fn write<T>(&mut self, iterator: T) -> Result<(), Self::Error>
    where
        T: Iterator<Item = RGB>;
}
```

### PR DESCRIPTION
**Pull request by david-sawatzke**
@evq @sajattack Do you have any other suggestions, especially for renaming `RGBA`?

Think this is otherwise ok
Closes #3

### CODE REVIEW

**Comment by evq** [comment]
hey @david-sawatzke. I think there are some advantages to making a `Pixel` associated type.

for one it seems to correspond well to how communication with LED strips generally works in that they have a concrete pixel type that they accept. often this will be RGB8, but some are RGBW8, RGB16, etc as we've discussed.

there are also strips where the color order is reversed (or otherwise is different) so the pixel type is BGR8. if one is trying to maximize performance it may be useful to precompute an animation in the strips native pixel type to avoid unnecessary channel swapping when outputting. what really matters is that you can convert the type you are writing into the strips native pixel type - which can be well 

another advantage is that for those writing code which attempts to generically wrap an LED strip, having an associated type cleans up the type signature. this results in end user benefits because common rust embedded frameworks like RTFM require writing the full type signature out when defining shared resources as far as I can tell.

**Comment by david-sawatzke** [comment]
Most situations where multiple trait implementations could be useful (e.g. being able to increase the pwm frequency) can also be handled by just implementing multiple types with their own `SmartLedsWrite` implementation. So making it associated doesn't do much harm in the edge cases and it certainly does clean up definitons, especially in cases where `PhantomData` might otherwise be needed.

I'm not happy about naming it `Pixel`, you could, for example, use multiple standalone ws2812 leds as indicators, where `Pixel` doesn't quite match. I've changed it to `Color`, think that should be (hopefully) clear enough.

**Comment by sajattack** [comment]
Yep, LGTM

File: src/lib.rs [hunk 1]
```diff
@@ -1,10 +1,17 @@
 #![no_std]
 
-pub type Color = rgb::RGB8;
+pub use {rgb::RGB16, rgb::RGB8, rgb::RGBA};
```
Comment (evq) [inline src/lib.rs]: maybe re-export `rgb:RGB` as well to keep both generic `RGB` and `RGBW` accessible in the same way?

Comment (david-sawatzke) [reply src/lib.rs]: Good catch

**Pull request merged**